src/libraries/CurrencySettleTake.sol

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import {Currency, CurrencyLibrary} from "../types/Currency.sol";
+import {IERC20Minimal} from "../interfaces/external/IERC20Minimal.sol";
+import {IPoolManager} from "../interfaces/IPoolManager.sol";
+
+library CurrencySettleTake {
+    using CurrencyLibrary for Currency;
+
+    /// @notice Settle (pay) a currency to the PoolManager
+    /// @dev Burning a native 6909 token takes precedence over a native currency transfer
```

Comment:
is there a particular reason burn takes precedent? idm either way was just curious

Reply:
if native had priority, none of the 6909 claims would work for native currency. i.e. you wouldnt be able to burn a 6909-wrapped native eth, since the if-condition would short circuit on currency.isNative() and use native transfers

was caught in one of the tests

Reply:
wonder if should update the code comment or just remove it entirely? maybe i change it to indicate that we want to support 6909-wrapped native currencies and we shouldnt short-circuit for native transfer?

Reply:
moved to the comment to within the function

Reply:
sweet looks good